cdap-common/src/main/java/io/cdap/cdap/common/guice/preview/PreviewDiscoveryRuntimeModule.java

```diff
@@ -38,8 +45,10 @@ protected void configure() {
     // and router can route directly to preview service. for other services we want to use local discovery service.
     bind(DiscoveryService.class).annotatedWith(
       Names.named("shared-discovery-service")).toInstance(discoveryService);
-    bind(InMemoryDiscoveryService.class).in(Singleton.class);
-    bind(DiscoveryService.class).to(InMemoryDiscoveryService.class);
-    bind(DiscoveryServiceClient.class).to(InMemoryDiscoveryService.class);
+    bind(DiscoveryServiceClient.class).annotatedWith(Names.named(PREVIEW_DISCOVERY)).toInstance(discoveryServiceClient);
```

Comment:
So this is pointing to the actual discovery service, right? I think it is better annotate it with `actualDiscoveryService`? (Similar to the `PreviewDataModules` naming).

Reply:
Fixed